docs/DeveloperGuide.adoc

```diff
@@ -279,104 +298,446 @@ Refer to the guide <<DevOps#, here>>.
+* 1a. The command specified by the user is not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid command error to user.
++
+_Use case ends_
+
+
+[discrete]
+=== Use case: Untag a transaction
 
-[[private-contact-detail]] Private contact detail::
-A contact detail that is not meant to be shared with others
+*MSS*
+
+1. User requests to untag a specific transaction.
+2. THRIFT processes the request and removes the tag from the specific transaction.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The parameters specified by the user are not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid parameter error to the user and the transaction is not updated.
++
+_Use case ends_
+
+[discrete]
+=== Use case: Convert a transaction
+
+*MSS*
+
+1. User requests to convert the currency of a specific transaction to a new currency.
+2. THRIFT processes the request and updates specific transaction with the new value in the new currency.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The parameters specified by the user are not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid parameter error to the user and the transaction is not updated.
++
+_Use case ends_
+
+[discrete]
+=== Use case: Convert a value
+
+*MSS*
+
+1. User requests to convert a value in SGD to a specific currency.
+2. THRIFT processes the request and displays the new value in the new currency.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The parameters specified by the user are not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid parameter error to the user.
++
+_Use case ends_
+
+[appendix]
+== Non Functional Requirements
+
+. THRIFT should be able to run on any <<mainstream-os, mainstream OS>> as long as it has `Java 11` installed.
+. THRIFT should be able to hold up to 250 transactions without a noticeable drop in performance for typical usage.
+. A user with above average typing speed for <<regular-english-text, regular English text>> (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+. THRIFT should be able to run with or without internet connection.
+. THRIFT should work for a single user only.
+. THRIFT should not require user to install.
+. Features implemented should be testable using manual testing and automated testing.
 
 [appendix]
-== Product Survey
+== Glossary
+
+[[thrift]] THRIFT::
+Stands for "Terminal-input Human Readable Interactive Finance Tracker". It is the application this developer guide is for.
 
-*Product Name*
+[[nfr]] NFR::
+Stands for "Non-functional Requirement"
 
-Author: ...
+[[mainstream-os]] Mainstream OS::
+Stands for commonly used Operating Systems (OS) such as Windows, Linux, Unix, OS-X
 
-Pros:
+[[regular-english-text]] Regular English Text::
+Stands for text with ordinary english grammar structures and vocabulary generally used by the public.
+It excludes syntax related to programming and <<system-administration, system administration>>.
```

Comment:
This exclusion is not necessary; the first sentence alone is sufficient. This also removes the need to further clarify the definition of 'System administration' further in the Glossary, and makes the Glossary more friendly to non-technical users.